src/governance.cpp

```diff
@@ -654,28 +659,23 @@ void CGovernanceManager::Sync(CNode* pfrom, uint256 nProp)
 
     {
         LOCK2(cs_main, cs);
+        object_m_it it = nProp == uint256() ? mapObjects.begin() : mapObjects.find(nProp);
```

Comment:
This code isn't quite right.  It will send mutliple objects when only one is requested.

It's also difficult to read.

Suggest something like:

`
object_m_it itBegin = mapObjects.begin();
object_m_it itEnd = mapObjects.end();
if(nProp != uint256()) {
  itBegin = mapObjects.find(nProp);
  if(itBegin == mapObjects.end()) {
    return;
  }
  itEnd = itBegin;
  ++itEnd;
}
...
for(object_m_it it = itBegin; it != itEnd; ++it) {
...
`